perfmetrics/scripts/testing_on_gke/examples/dlio/dlio_workload.py

```diff
@@ -0,0 +1,118 @@
+"""This file defines a DlioWorkload (a DLIO Unet3d workload) and provides utility for parsing a json
+
+test-config file for a list of them.
+"""
+
+import json
+
+
+def validateDlioWorkload(workload: dict, name: str):
+  """Validates the given json workload object."""
+  if 'dlioWorkload' not in workload:
+    print(f"{name} does not have 'dlioWorkload' key in it.")
+    return False
+
+  if 'bucket' not in workload:
+    print(f"{name} does not have 'bucket' key in it.")
+    return False
+
+  if 'fioWorkload' in workload:
+    print(f"{name} has 'fioWorkload' key in it, which is unexpected.")
+    return False
+
+  dlioWorkload = workload['dlioWorkload']
+  for requiredAttribute, expectedType in {
+      'numFilesTrain': int,
+      'recordLength': int,
+      'batchSizes': list,
+  }.items():
+    if requiredAttribute not in dlioWorkload:
+      print(
+          f'In {name}, dlioWorkload for {name} does not have'
+          f' {requiredAttribute} in it'
+      )
+      return False
+    if not type(dlioWorkload[requiredAttribute]) is expectedType:
+      print(
+          f'In {name}, dlioWorkload[{requiredAttribute}] is of type'
+          f' {type(dlioWorkload[requiredAttribute])}, expected:'
+          f' {expectedType} '
+      )
+      return False
+
+  for batchSize in dlioWorkload['batchSizes']:
+    if not type(batchSize) is int:
+      print(
+          f'In {name}, one of the batch-size values in'
+          f" dlioWorkload['batchSizes'] is '{batchSize}', which is of type"
+          f' {type("batchSize")}, not int'
+      )
+      return False
+    if batchSize < 1:
+      print(
+          f'In {name}, one of the batch-size values in'
+          f" dlioWorkload['batchSizes'] is '{batchSize}' < 1, which is not"
+          ' supported.'
+      )
+      return False
+
+  return True
+
+
+class DlioWorkload:
+  """DlioWorkload holds data needed to define a DLIO Unet3d workload
+
+  (essentially the data needed to create a job file for DLIO run).
+
+  Members:
+  1. scenario (string): One of "local-ssd", "gcsfuse-generic",
+  "gcsfuse-file-cache" and "gcsfuse-no-file-cache".
+  2. numFilesTrain (string): DLIO numFilesTrain argument in string format e.g.
+  '500000' etc.
+  3. recordLength (string): DLIO recordLength argument in string format e.g.
+  '100', '1000000', '10M' etc.
+  4. bucket (string): Name of a GCS bucket to read input files from.
+  5. batchSizes (list of strings): a string containing multiple comma-separated
+  integer values.
+  """
+
+  def __init__(self, scenario, numFilesTrain, recordLength, bucket, batchSizes):
+    self.scenario = scenario
+    self.numFilesTrain = numFilesTrain
+    self.recordLength = recordLength
+    self.bucket = bucket
+    self.batchSizes = batchSizes
```

Comment:
Switch self.batchSizes to set to avoid duplicates.

```suggestion
    self.batchSizes = set(batchSizes)
```

Reply:
done